test/Microsoft.AspNet.Mvc.Extensions.Test/ModelBinding/Metadata/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -447,29 +448,11 @@ public void GetDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool exp
                         // Note order duplicates appear cannot be inferred easily e.g. does not match the source.
                         // Zero is before None but Two is before Duece in the class below.
                         typeof(EnumWithDuplicates),
-                        new List<KeyValuePair<string, string>>
-                        {
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Zero), "0"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.None), "0"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.One), "1"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Duece), "2"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Two), "2"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.MoreThanTwo), "3"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Three), "3"),
-                        }
+                        GetValuesListUptoThree()
                     },
                     {
                         typeof(EnumWithDuplicates?),
-                        new List<KeyValuePair<string, string>>
-                        {
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Zero), "0"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.None), "0"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.One), "1"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Duece), "2"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Two), "2"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.MoreThanTwo), "3"),
-                            new KeyValuePair<string, string>(nameof(EnumWithDuplicates.Three), "3"),
-                        }
+                        GetValuesListUptoThree()
```

Comment:
Look here: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Extensions/ModelBinding/Metadata/DataAnnotationsMetadataProvider.cs#L113

This is undefined behavior and can change at any point. The test shouldn't hardcode this, it should be tolerant.